src/core/core.controller.js

```diff
@@ -918,6 +928,25 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 		}
 	},
 
+	/**
+	 * @private
+	 */
+	_updateHoverStyles: function() {
+		var me = this;
+		var options = me.options || {};
+		var hoverOptions = options.hover;
+
+		// Remove styling for last active (even if it may still be active)
+		if (me.lastActive.length) {
+			me.updateHoverStyle(me.lastActive, hoverOptions.mode, false);
+		}
+
+		// Built-in hover styling
+		if (me.active.length && hoverOptions.mode) {
```

Comment:
Is it right that we would sometimes `updateHoverStyle` twice in this method? Or should we change it to the code below?

```
		if (me.active.length && hoverOptions.mode) {
			me.updateHoverStyle(me.active, hoverOptions.mode, true);
		} else if (me.lastActive.length) {
			me.updateHoverStyle(me.lastActive, hoverOptions.mode, false);
		}
```

Reply:
First call removes styles from previously hovered items, second updates newly hovered.